### DISCUSSION TITLE
possibility to provide custom validations for link/image url

### DISCUSSION THREAD

**Discussion by Maks-Yaremenko** [sibiraj-s/ngx-editor#312] [Q&A] [answered]
It would be great to add custom validation for image/link URL, currently, some valid URLs doesn't pass default validation.

**Comment by sibiraj-s** [comment, marked as answer]
Menu component is supposed to be auxiliary. I am not sure adding/providing custom validators or more function will help always. Since the requirement changes always.

I have exposed commands in [v6.1.0-beta.0](https://github.com/sibiraj-s/ngx-editor/releases/tag/v6.1.0-beta.0). with that you should be able to insertLink

```ts
this.editor.commands
  .insertImage("https://<YOUR_IMAGE_URL>")
  .exec();
```

and by using [customMenu](https://sibiraj-s.github.io/ngx-editor/#/menu?id=custom-menu) you should be able to create/apply whatever functions needed with ease.

**Comment by d-badia** [comment]
As @Maks-Yaremenko said, some valid URLs doesn't pass default validation. So, would there be any possibility to disable validation?

**Reply by sibiraj-s** [reply]
As Mentioned earlier. You can create custom menu items for custom validation or even disabling it. I am open to updating the regex though to support more url patterns.

[a comment by an automated account is not shown]
